src/main/java/org/apache/sysml/hops/ipa/IPAPassRemoveUnnecessaryCheckpoints.java

```diff
@@ -214,16 +214,17 @@ private static void removeCheckpointReadWrite(DMLProgram dmlp)
 		throws HopsException
 	{
 		List<StatementBlock> sbs = dmlp.getStatementBlocks();
-		
-		if( sbs.size()==1 & !(sbs.get(0) instanceof IfStatementBlock 
-			|| sbs.get(0) instanceof WhileStatementBlock 
-			|| sbs.get(0) instanceof ForStatementBlock) ) 
-		{
-			//recursively process all dag roots
-			if( sbs.get(0).getHops()!=null ) {
-				Hop.resetVisitStatus(sbs.get(0).getHops());
-				for( Hop root : sbs.get(0).getHops() )
-					rRemoveCheckpointReadWrite(root);
+
+		if(!sbs.isEmpty()) {
+			if (sbs.size() == 1 & !(sbs.get(0) instanceof IfStatementBlock
```

Comment:
Why is this extra condition required? Isn't it redundant to the check for `sbs.size() == 1`?

Reply:
I am not sure, this seems to be there from the beginning.

Reply:
Didn't you add the `if(!sbs.isEmpty()) {`?

Reply:
Yes, I did. To avoid `NullPointerException`

Reply:
ok now I see it - please simply change "sbs.size() == 1 &" to "sbs.size() == 1 &&" and remove the new redundant check for "if(!sbs.isEmpty()) {" (i.e., if size==1 is guaranteed that it's not empty).